virt-launcher pod fails to start because of SELinux problems

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
Fail to start a VM on the environment with enabled SELinux
virt-launcher log
```
level=error timestamp=2018-06-21T10:11:09.670507Z pos=server.go:154 component=virt-launcher reason="listen unix /var/run/kubevirt/sockets/kubevirt-test-default_testvmi64qfv_sock: bind: permission denied" msg="failed to create unix socket for launcher cmd service"
level=error timestamp=2018-06-21T10:11:09.670570Z pos=virt-launcher.go:81 component=virt-launcher reason="listen unix /var/run/kubevirt/sockets/kubevirt-test-default_testvmi64qfv_sock: bind: permission denied" msg="Failed to start virt-launcher cmd server"
panic: listen unix /var/run/kubevirt/sockets/kubevirt-test-default_testvmi64qfv_sock: bind: permission deniedleve
```

**What you expected to happen**:
A VM must succeed to start without any SELinux errors

**How to reproduce it (as minimally and precisely as possible)**:
- start a VM on the environment with enabled SELinux

**Anything else we need to know?**:
SELinux errors
```
ausearch -m avc --start recent
----
time->Thu Jun 21 12:03:19 2018
type=PROCTITLE msg=audit(1529582599.229:10379): proctitle="/usr/sbin/libvirtd"
type=SYSCALL msg=audit(1529582599.229:10379): arch=c000003e syscall=257 success=no exit=-13 a0=ffffffffffffff9c a1=7f164122484e a2=80001 a3=0 items=0 ppid=20653 pid=20668 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="libvirtd" exe="/usr/sbin/libvirtd" subj=system_u:system_r:container_t:s0:c67,c240 key=(null)
type=AVC msg=audit(1529582599.229:10379): avc:  denied  { write } for  pid=20668 comm="libvirtd" name="mtrr" dev="proc" ino=4026531960 scontext=system_u:system_r:container_t:s0:c67,c240 tcontext=system_u:object_r:mtrr_device_t:s0 tclass=file
----
time->Thu Jun 21 12:03:28 2018
type=PROCTITLE msg=audit(1529582608.902:10398): proctitle=2E2F766972742D6C61756E63686572002D2D71656D752D74696D656F757400356D002D2D6E616D6500766D692D6665646F7261002D2D6E616D6573706163650064656661756C74002D2D6B756265766972742D73686172652D646972002F7661722F72756E2F6B75626576697274002D2D72656164696E6573732D66696C65
type=SYSCALL msg=audit(1529582608.902:10398): arch=c000003e syscall=49 success=no exit=-13 a0=7 a1=c4201b8410 a2=34 a3=0 items=0 ppid=20578 pid=20616 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="virt-launcher" exe="/virt-launcher" subj=system_u:system_r:container_t:s0:c67,c240 key=(null)
type=AVC msg=audit(1529582608.902:10398): avc:  denied  { write } for  pid=20616 comm="virt-launcher" name="sockets" dev="tmpfs" ino=112881 scontext=system_u:system_r:container_t:s0:c67,c240 tcontext=system_u:object_r:container_var_run_t:s0 tclass=dir
```

**Environment**:
```
oc v3.10.0-rc.0+c20e215
kubernetes v1.10.0+b81c8f8
features: Basic-Auth GSSAPI Kerberos SPNEGO
openshift v3.10.0-rc.0+c20e215
kubernetes v1.10.0+b81c8f8
```

As a data point, this issue really is limited to SELinux (as opposed to being compounded by other issues). I temporarily turned off SElinux enforcement on my node and my VM instance booted fine.

audit2allow suggests adding these rules:

```

module local_rules 1.0;

require {
        type mtrr_device_t;
        type container_var_run_t;
        type container_t;
        class sock_file create;
        class dir { add_name write };
        class file write;
}

#============= container_t ==============

allow container_t container_var_run_t:dir { add_name write };
allow container_t container_var_run_t:sock_file create;

allow container_t mtrr_device_t:file write;
```

However, after adding those I'm still getting a failure in the pod's logs:

```
level=info timestamp=2018-06-21T16:49:44.168216Z pos=libvirt.go:256 component=virt-launcher msg="Connecting to libvirt daemon: qemu:///system"
level=info timestamp=2018-06-21T16:49:54.171291Z pos=libvirt.go:271 component=virt-launcher msg="Connected to libvirt daemon"
level=error timestamp=2018-06-21T16:49:54.171822Z pos=server.go:154 component=virt-launcher reason="listen unix /var/run/kubevirt/sockets/default_vmi-ephemeral_sock: bind: permission denied"
msg="failed to create unix socket for launcher cmd service"              
level=error timestamp=2018-06-21T16:49:54.171901Z pos=virt-launcher.go:81 component=virt-launcher reason="listen unix /var/run/kubevirt/sockets/default_vmi-ephemeral_sock: bind: permission de
nied" msg="Failed to start virt-launcher cmd server"                     
panic: listen unix /var/run/kubevirt/sockets/default_vmi-ephemeral_sock: bind: permission denied
                                       
goroutine 1 [running]:                                       
main.startCmdServer(0xc420043f40, 0x34, 0x12800e0, 0xc4200d0290, 0xc420096de0, 0xc42087de5f)
        /root/go/src/kubevirt.io/kubevirt/cmd/virt-launcher/virt-launcher.go:82 +0x446
main.main()                                                 
        /root/go/src/kubevirt.io/kubevirt/cmd/virt-launcher/virt-launcher.go:266 +0x569
virt-launcher exited with code 2
```

--- a/pkg/virt-controller/services/template.go
+++ b/pkg/virt-controller/services/template.go
@@ -323,6 +323,9 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 			Subdomain: vmi.Spec.Subdomain,
 			SecurityContext: &k8sv1.PodSecurityContext{
 				RunAsUser: &userId,
+				SELinuxOptions: &k8sv1.SELinuxOptions{
+					Type: "spc_t",
+				},
 			},
 			TerminationGracePeriodSeconds: &gracePeriodKillAfter,
 			RestartPolicy:                 k8sv1.RestartPolicyNever,

